src/main/java/org/openrewrite/staticanalysis/TernaryOperatorsShouldNotBeNested.java

```diff
@@ -0,0 +1,368 @@
+                        : bin.getLeft();
+            } else {
+                throw new IllegalArgumentException(
+                        "Only J.Binary or J.MethodInvocation are expected as ternary conditions when creating a switch case");
+            }
+            return new J.Case(
+                    Tree.randomId(),
+                    ternary.getPrefix().withWhitespace(" "),
+                    ternary.getMarkers(),
+                    J.Case.Type.Rule,
+                    JContainer.build(
+                            Collections.singletonList(JRightPadded.<Expression>build(compare.withPrefix(Space.SINGLE_SPACE))
+                                    .withAfter(Space.SINGLE_SPACE))
+                    ),
+                    JContainer.build(Collections.emptyList()),
+                    JRightPadded.build(ternary.getTruePart())
+            );
+        }
+
+        private J.Case toDefault(final J.Ternary ternary) {
+            return new J.Case(
+                    Tree.randomId(),
+                    Space.EMPTY,
+                    ternary.getMarkers(),
+                    J.Case.Type.Rule,
+                    JContainer.build(Collections.singletonList(JRightPadded.<Expression>build(new J.Identifier(
+                            randomId(),
+                            Space.EMPTY,
+                            Markers.EMPTY,
+                            "default",
+                            null,
+                            null
+                    )).withAfter(Space.SINGLE_SPACE))),
+                    JContainer.build(Collections.emptyList()),
+                    JRightPadded.build(ternary.getFalsePart())
+            );
+        }
+
+        private Optional<J.Identifier> findConditionIdentifier(final J.Ternary ternary) {
+            J.Identifier result = null;
+            if (ternary.getCondition() instanceof J.MethodInvocation) {
+                J.MethodInvocation inv = (J.MethodInvocation) ternary.getCondition();
+                if (!inv.getSimpleName().equals("equals")) {
+                    return Optional.empty();
+                }
+                if (isVariable(inv.getSelect())) {
+                    result = (J.Identifier) inv.getSelect();
+                }
+                if (inv.getArguments().size() == 1 && inv.getArguments().get(0) instanceof J.Identifier) {
+                    result = (J.Identifier) inv.getArguments().get(0);
+                }
+                if (isObjectsEquals(inv)) {
+                    //one has to be constant, other not
+                    J first = inv.getArguments().get(0);
+                    J second = inv.getArguments().get(1);
+                    result = xorVariable(first, second);
+                }
+            }
+            if (ternary.getCondition() instanceof J.Binary) {
+                J.Binary bin = (J.Binary) ternary.getCondition();
+                result = xorVariable(bin.getLeft(), bin.getRight());
```

Comment:
In the body of this `if` we need to check the `J.Binary#getOperator()` result. I think currently we only support `==` (and not `!=`), so we should make sure that this is indeed also the case. A corresponding test would also be nice.

Reply:
good catch, I will extend!